### DISCUSSION TITLE
cacheModels config question

### DISCUSSION THREAD

**Discussion by desi-dimitrova** [vladmandic/human#255] [General]
Hi Vlad,

If I set the cacheModels configuration to FALSE, it still saves and loads the models from the IndexedDB. I think that may be a bug but I am not sure, so I wanted to ask what is the expected behavior when it's set to FALSE? I'm using the latest version 2.6.4.
![image](https://user-images.githubusercontent.com/98762671/158582662-2198c464-4fa2-4e0d-8862-27d8ca08e3a7.png)

When using version 2.5.8, some users reported an error regarding loading the models, see screenshot:
![image (15)](https://user-images.githubusercontent.com/98762671/158583537-996df7e4-eab4-4a59-8e57-deefade45bb8.png)
I am not sure if this is still an issue and it's not something that can be reproduced easily, it happens rarely.  I was hoping to have a way to disable loading the models from the IndexedDB in case I run into this issue again, that is why I'm asking if setting cacheModels to false will help with that. The weirdest part from getting this error is that when I reproduced it I had all models available in a folder and the IndexedDB as well...

**Comment by vladmandic** [comment]
i've created an issue #256 for this as its an actual bug (and i've just fixed it, see the issue for details)

**Reply by desi-dimitrova** [reply]
Thank you!! :)

**Comment by desi-dimitrova** [comment]
Out of curiosity, do you know in what situation this error shows up? I can't really pin-point it because I've seen it when all the models I'm using are available in a models folder and are saved in the indexedDB both on PC & Mac.

**Comment by vladmandic** [comment]
first time each model gets loaded from folder and stored in browser indexdb for use as future cache.
next time, human check if model is registered in cache and if yes it tries to load it from browser indexdb and if its not registered, it will load from folder.

in this case, seems like model somehow got registered in cache, but actual cache content is empty.  
so either:
- something strange (unaccounted for) went wrong with previous save to cache or
- something emptied the cache in-between when human looked it up and tried to load model

additional info can be gathered by enabling `config.debug = true`, but that would get really verbose really fast.

**Reply by desi-dimitrova** [reply]
I see, thanks!

**Comment by vladmandic** [comment]
btw, if you manage to reproduce, let me know - i'd like to handle any exceptions in the code.
